IOPool/Output/src/RootOutputFile.cc

```diff
@@ -608,11 +612,40 @@ namespace edm {
     b->Fill();
   }
 
+  // For duplicate removal and to determine if fast cloning is possible, the input
+  // module by default reads the entire EventAuxiliary branch when it opens the
+  // input files.  If EventAuxiliary is written in the usual way, this results
+  // in many small reads scattered throughout the file, which can have very poor
+  // performance characteristics on some filesystems.  As a workaround, we save
+  // EventAuxiliary data and the ProcessHistoryID, writing the branch as we are
+  // closing the file, resulting in (hopefully) one basket at the end of the file
+  // that can be accessed with one read.
+  void RootOutputFile::writeEventAuxiliary() {
+    auto tree = eventTree_.tree();
+    auto bname = BranchTypeToAuxiliaryBranchName(InEvent).c_str();
+
+    tree->SetBranchStatus(bname, true);
+    auto basketsize = eventAuxiliaryVector_.size()*(sizeof(EventAuxiliary)+26); // 26 is an empirical fudge factor
+    tree->SetBasketSize(bname, basketsize);
+    auto b = tree->GetBranch(bname);
+
+    assert(tree->GetEntries() == static_cast<Long64_t>(eventAuxiliaryVector_.size()));
```

Comment:
What happens now if we have a job that does fast cloning but then terminates early? Will this assert go off? If so, that shouldn't be an assert and instead should probably be an exception.

Reply:
Could you add a unit test to the system which has such a behavior?